src/paymentwall_pingback.php

```diff
@@ -2,19 +2,42 @@
 
 require 'paymentwall_init.php';
 
+include(DIR_WS_CLASSES . 'order.php');
+
 $pingback = new Paymentwall_Pingback($_GET, $_SERVER['REMOTE_ADDR']);
+
 if ($pingback->validate()) {
+
     $productId = $pingback->getProduct()->getId();
-    $id = tep_db_query("select orders_id from " . TABLE_ORDERS . " where orders_id = '" . (int)$productId . "'");
+    $order = new order($productId);
+    $_info = array_filter($order->info);
 
-    if (mysqli_fetch_all($id)[0][0]) {
+    if (!empty($_info)) {
         if ($pingback->isDeliverable()) {
+
+            if (MODULE_PAYMENT_PAYMENTWALL_DELIVERY == 'True') {
+                $delivery = new Paymentwall_GenerericApiObject('delivery');
+                $response = $delivery->post(getDeliveryConfirmationData($order, $pingback->getReferenceId()));
+            }
```

Comment:
This feature can enable or disable.
In OSCommerce, the settings store in the DB and retrieve through constant variables.